### PR TITLE
Add network scanning, move communication to storage

### PR DESCRIPTION
**Pull request by atjn**
Instead of using message channels, we now communicate through storage.

The plugin knows what it is doing via the `ScanState` class, which is a state machine that saves its state to storage.

I have added some basic network scanning capabilities to the background script. 

I have also added flags on each plugin so we know if each script needs access to the DOM and/or network info. In a future PR, I will disable the collection of them if no selected plugin requires them.

### CODE REVIEW

**Comment by BaBrixx** [comment]
>  In a future PR, I will disable the collection of them if no selected plugin requires them.

Consider creating an issue for this if you haven't done it already

File: source/lib/pluginTypes.ts [hunk 1]
```diff
@@ -6,8 +6,27 @@ export type PluginInput = {
...
+	/**
+	 * The plugin name.
+	 */
 	readonly name: string;
+	/**
+	 * Versioning of each individual plugin.
+	 */
 	readonly version: string;
+	/**
+	 * If true, means we need to scan the contents fo the DOM
```
Comment (DenFlyvendeGed) [inline source/lib/pluginTypes.ts:18]: of
```suggestion
	 * If true, means we need to scan the contents fo the DOM
```

**Review by BaBrixx — DISMISSED** [review]
lgtm

File: source/background.ts [hunk 2]
```diff
@@ -3,53 +3,94 @@ import { detect } from "detect-browser";
...
-			} = SendContentSchema.safeParse(request);
-			if (!requestSuccess) {
-				debug.warn(requestError);
-				return;
+const listeners: Array<keyof typeof browser.webRequest> = [
+	"onBeforeRequest",
+	"onBeforeRedirect",
+	"onCompleted",
+	"onErrorOccurred",
+];
```
Comment (BaBrixx) [inline source/background.ts:21]: Maybe add comment

Comment (atjn) [reply source/background.ts:21]: hello this is a comment

**Review by BaBrixx — DISMISSED** [review]
Godt at du helt selv fandt på den kommentar

**Review by BaBrixx — APPROVED** [review]
I kill you

**Pull request merged**